### ISSUE TITLE
Modifying user permission failed when an optional custom field is empty

### ISSUE THREAD

**Issue by lonelyuan** [CTFd/CTFd#1653]
**Environment**:

- CTFd Version/Commit: 3.1.0-release
- Operating System: ubuntu18.04
- Web Browser and Version: chrome 85.0

**What happened?**

I set an optional custom field for my CTFd, I found when it's empty, I can't modify user permission. The submit button have no response. 
Here are the error log in console:
```
vendor.bundle.min.js?d=b4adfef4:150 Uncaught Error: Syntax error, unrecognized expression: :input[name=fields[3]]
    at Function.se.error (vendor.bundle.min.js?d=b4adfef4:150)
    at se.tokenize (vendor.bundle.min.js?d=b4adfef4:150)
    at se.select (vendor.bundle.min.js?d=b4adfef4:150)
    at Function.se [as find] (vendor.bundle.min.js?d=b4adfef4:150)
    at k.fn.init.find (vendor.bundle.min.js?d=b4adfef4:150)
    at eval (utils.js:43)
    at Array.map (<anonymous>)
    at k.fn.init._jquery.default.fn.serializeJSON (utils.js:38)
    at HTMLFormElement._ (user.min.js?d=b4adfef4:1)
    at HTMLFormElement.dispatch (vendor.bundle.min.js?d=b4adfef4:150)
se.error @ vendor.bundle.min.js?d=b4adfef4:150
se.tokenize @ vendor.bundle.min.js?d=b4adfef4:150
se.select @ vendor.bundle.min.js?d=b4adfef4:150
se @ vendor.bundle.min.js?d=b4adfef4:150
find @ vendor.bundle.min.js?d=b4adfef4:150
eval @ utils.js:43
_jquery.default.fn.serializeJSON @ utils.js:38
_ @ user.min.js?d=b4adfef4:1
dispatch @ vendor.bundle.min.js?d=b4adfef4:150
g.handle @ vendor.bundle.min.js?d=b4adfef4:150
```
When I set some value to that empty field, everything is all right. But it should be optional.

**Comment by ColdHeat** [comment]
Can you be a little more descriptive? What page is this happening on?

**Comment by lonelyuan** [comment]
Sorry for that. It's in admin panel, page "users", clicking "Edit user". 
Here is screenshot:
![20200917225754](https://user-images.githubusercontent.com/44459536/93490093-a16e4800-f93a-11ea-82d7-f2052868f307.png)
 I found this bug when I want to give admin permission to some accounts. Moreover , I found when a custom field is empty, no editing operation to that user can be done, though that field has been set optional. When I set something for that field, everything is fine. So I guess JS code here have a little bug.

**Comment by ColdHeat** [comment]
Hmm I can reproduce this on demo but I can't reproduce this locally. Do you know what commit you're using?

**Comment by ColdHeat** [comment]
Okay I think I see this issue. I think the minifer is stripping out some important quotes here which it shouldn't be doing. This sounds like a bug in whichever minifier we're using.

**Comment by ColdHeat** [comment]
It looks like an updated version of @babel/core fixes this. 

You can run the following to fix this in the meantime. 

```
yarn add "@babel/core" --latest
yarn build
```

**Comment by AXDOOMER** [comment]
This issue also affects custom fields on the users' settings page.

**Comment by ColdHeat** [comment]
@AXDOOMER do the above commands fix it for you? I can merge in #1654 in but I'm not sure if it will break something else.
